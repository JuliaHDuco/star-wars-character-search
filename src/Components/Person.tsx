import * as React from 'react';
import { IPerson } from '../types';
import './styling/Person.scss';
import Modal from './Modal';

interface IPersonProps {
    info: IPerson;
    onHide: (e: React.MouseEvent) => void;
}

const Person = ({info, onHide}: IPersonProps) => {
  const formatKey = (key: string) => {
    let formattedKey = key.replace(/^\w/, (character) => character.toUpperCase());

    formattedKey = formattedKey.replace('_', ' ');
    return formattedKey;
  }

  const keysToShow = [
    'name',
    'height',
    'mass',
    'hair_color',
    'skin_color',
    'eye_color',
    'birth_year',
    'gender',
  ]

  return (
    <Modal onHide={onHide}>
      <div className='person'>
        <ul>
          {
            Object.keys(info).map((key: string) => {
              if (keysToShow.includes(key)) {
                return (
                  <li key={key}>
                    {formatKey(key)}: {info[key]}
                  </li>)
              }

              return null;
            })
          }
        </ul>
      </div>
    </Modal>
  );
}

export default Person;
