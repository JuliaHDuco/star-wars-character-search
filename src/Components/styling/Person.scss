.person {
  ul {
    margin-top: 1em;

    li {
      width: 50%;
    }
  }
}
