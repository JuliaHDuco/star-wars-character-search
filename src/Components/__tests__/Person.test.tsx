import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Person from '../Person';
import {testPerson} from '../../utilities/tests/data/person';

test('should render Person', () => {
  const onHide = jest.fn();
  const component = renderer.create(
    <Person onHide={onHide} info={testPerson} />
  ).toJSON();
  expect(component).toMatchSnapshot();
});
