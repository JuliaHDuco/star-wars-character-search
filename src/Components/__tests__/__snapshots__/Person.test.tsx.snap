// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Person 1`] = `
<div
  className="modal-wrapper"
>
  <div
    className="modal"
  >
    <button
      onClick={[MockFunction]}
    />
    <div
      className="person"
    >
      <ul>
        <li>
          Name
          : 
          Luke Skywalker
        </li>
        <li>
          Height
          : 
          172
        </li>
        <li>
          Mass
          : 
          77
        </li>
        <li>
          Hair color
          : 
          blond
        </li>
        <li>
          Skin color
          : 
          fair
        </li>
        <li>
          Eye color
          : 
          blue
        </li>
        <li>
          Birth year
          : 
          19BBY
        </li>
        <li>
          Gender
          : 
          male
        </li>
      </ul>
    </div>
  </div>
</div>
`;
